cpp/cmake_modules/ThirdpartyToolchain.cmake

```
@@ -1057,7 +1057,12 @@ macro(build_boost)
   set(BOOST_VENDORED TRUE)
 endmacro()
 
-if(ARROW_BUILD_TESTS)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" AND CMAKE_CXX_COMPILER_VERSION
+                                              VERSION_GREATER_EQUAL 16)
```

We can't use `VERSION_GREATER_EQUAL` for now because it's available since CMake 3.7:
https://cmake.org/cmake/help/latest/command/if.html#version-greater-equal

---

How about this then?

```suggestion
                                              VERSION_GREATER 15)
```

---

We can use it.